core/src/main/scala/org/apache/spark/status/protobuf/Utils.scala

```diff
@@ -17,16 +17,18 @@
 
 package org.apache.spark.status.protobuf
 
-import com.google.protobuf.MessageOrBuilder
-
 object Utils {
   def getOptional[T](condition: Boolean, result: () => T): Option[T] = if (condition) {
     Some(result())
   } else {
     None
   }
 
-  def setStringField(input: String, f: String => MessageOrBuilder): Unit = {
+
+  // TODO(SPARK-42147): Restore the signature of `f` when `f` is declared as
+  //  `f: String => MessageOrBuilder` and `setStringField` used by other modules
+  //  not cause maven build failure
+  def setStringField(input: String, f: String => Any): Unit = {
```

Comment:
If the return value of f() doesn't matter, then this signature seems fine? no need to change it later
Did some change actually fail to compile as a result in master?

Reply:
OK, Let me remove this TODO.
Yes, some pr under development has this issue